_posts/2020-02-7-dns-based-discovery.md

```diff
@@ -0,0 +1,94 @@
+
+So in summary, we have a static list that provides us with nodes we can connect to which then allows us to synchronize the rest of the network using something like Kademlia. But what we need is something that can easily be mutated, guarantees a certain amount of security, and is efficient for resource restricted devices. Ideally our solution would also be robust and scalable.
+
+How do we do this? 
+
+Let's talk about a neat proposal, [EIP 1459: Node Discovery via DNS](https://eips.ethereum.org/EIPS/eip-1459), which is one of the strategies we are using for discovering waku nodes. [EIP-1459](https://eips.ethereum.org/EIPS/eip-1459) is a DNS-based discovery protocol that stores merkle trees in DNS records which contain connection information for nodes.
+
+Waku is our fork of Whisper. Oskar recently wrote an [entire post](https://vac.dev/fixing-whisper-with-waku) explaining it recently. In short, Waku is our method of fixing the shortcomings of Whisper in a more iterative fashion. 
+
+DNS-based methods for bootstrapping p2p networks are quite popular. Even bitcoin uses it, but it uses a concept called DNS seeds, which are just DNS servers that are configured to return a list of randomly selected nodes from the network upon being queried. This means that although these seeds are hardcoded in the client, the IP addresses of actual nodes do not have to be.
+
+```console
+> dig dnsseed.bluematt.me +short
+129.226.73.12
+107.180.78.111
+169.255.56.123
+91.216.149.28
+85.209.240.91
+66.232.124.232
+207.55.53.96
+86.149.241.168
+193.219.38.57
+190.198.210.139
+74.213.232.234
+158.181.226.33
+176.99.2.207
+202.55.87.45
+37.205.10.3
+90.133.4.73
+176.191.182.3
+109.207.166.232
+45.5.117.59
+178.211.170.2
+160.16.0.30
+```
+
+The above displays the result of querying on of these DNS seeds. All the nodes are stored as [`A` records](https://simpledns.plus/help/a-records) for the given domain name. This is quite a simple solution which bitcoin almost soley relies on since removing the [IRC bootstrapping method in v0.8.2](https://en.bitcoin.it/wiki/Network#IRC).
+
+So what makes this DNS based discovery useful? Well, it allows us to have a mutable list of bootstrap nodes without needing to ship a new version of the client every time a list is mutated. It also allows for a more lightweight method of discovering nodes, something very important for resource restricted devices.
+
+Additionally, DNS provides us with a robust and scalable infrastructure. This is due to its hierarchical architecutre. This hierarchical architecture also already makes it distributed such that the failure of one DNS server does not result in us no longer being able to resolve our name.
```

Comment:
I'd say it is battle tested and used by <the whole Internet>

architecute typoe